common/src/com/thoughtworks/go/server/websocket/Message.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2015 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.websocket;
+
+import com.thoughtworks.go.util.SystemEnvironment;
+import org.eclipse.jetty.websocket.api.WebSocketPolicy;
+
+import java.io.*;
+
+public class Message {
+
+    public static void setupPolicy(WebSocketPolicy policy, SystemEnvironment environment) {
+        policy.setMaxTextMessageBufferSize(environment.getWebsocketMaxMessageSize());
+        policy.setMaxTextMessageSize(environment.getWebsocketMaxMessageSize());
+        policy.setMaxBinaryMessageSize(environment.getWebsocketMaxMessageSize());
+        policy.setMaxBinaryMessageBufferSize(environment.getWebsocketMaxMessageSize());
+    }
+
+    public static byte[] encode(Message msg) {
+        String encode = JsonMessage.encode(msg);
```

Comment:
I don't have prefer to String or byte[].
The reason I choose byte[] instead of String is:
I tried to add GZip to JSON message communication, because there is message size limit in websocket communication, and I worried 1mb limit size may have problem when user configured a job has lots of commands or materials stuff. When we serialize BuildWork, it contains a lot of stuff, I can't predict how much data it may include. We may need long time to find out all unnecessary large data fields (like Modification#files @wpc found yesterday), and exclude them from websocket message. Since GZip will need convert String into byte[], I don't think we need convert them back to String again, because in lower level communication, String will be converted to byte[] again.
I pulled out GZip, because I feel it adds complexity and it won't solve root cause. 
Since I don't have prefer, so left byte[] here.